internal/libtor/enabled.go

```diff
@@ -0,0 +1,277 @@
+//go:build ooni_libtor
+
+package libtor
+
+// Adapted from https://github.com/cretz/bine
+// SPDX-License-Identifier: MIT
+
+//
+// #cgo linux,amd64 CFLAGS: -I${SRCDIR}/linux/amd64/include
+// #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/linux/amd64/lib -ltor -levent -lssl -lcrypto -lz -lm
+//
+// #cgo android,arm CFLAGS: -I${SRCDIR}/android/arm/include
+// #cgo android,arm LDFLAGS: -L${SRCDIR}/android/arm/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,arm64 CFLAGS: -I${SRCDIR}/android/arm64/include
+// #cgo android,arm64 LDFLAGS: -L${SRCDIR}/android/arm64/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,386 CFLAGS: -I${SRCDIR}/android/386/include
+// #cgo android,386 LDFLAGS: -L${SRCDIR}/android/386/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,amd64 CFLAGS: -I${SRCDIR}/android/amd64/include
+// #cgo android,amd64 LDFLAGS: -L${SRCDIR}/android/amd64/lib -ltor -levent -lssl -lcrypto -lz
+//
+// #include <limits.h>
+// #include <stdbool.h>
+// #include <stdlib.h>
+//
+// #include <tor_api.h>
+//
+// /* Note: we need to define inline helpers because we cannot index C arrays in Go. */
+//
+// static char **cstringArrayNew(size_t size) {
+//     char **argv = calloc(sizeof(char *), size);
+//     if (argv == NULL) {
+//         abort();
+//     }
+//     return argv;
+// }
+//
+// static void cstringArraySet(char **argv, size_t index, char *entry) {
+//     argv[index] = entry;
+// }
+//
+// static void cstringArrayFree(char **argv, size_t size) {
```

Comment:
Did you think about adding some safety checks here? 
I do understand that the way you use these functions, argv and the entries will not be NULL.

Reply:
`free(NULL)` is working as intended in virtually all systems nowadays. However, historically, that was a portability concern, but it seems this is not a concern anymore. (See https://github.com/maxmind/geoip-api-c/pull/100#pullrequestreview-143264986 for a discussion on this topic.)

Reply:
Got it, thanks for clarifying!